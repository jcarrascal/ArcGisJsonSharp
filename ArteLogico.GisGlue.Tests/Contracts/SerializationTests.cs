using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ArteLogico.GisGlue.Contracts;
using Newtonsoft.Json;
using NUnit.Framework;

namespace ArteLogico.GisGlue.Tests.Contracts
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("SerializationTestsApprovals")]
    public class SerializationTests
    {
        [Test]
        public void FeatureServiceOperationalLayersRoundtripSerialization()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/ArcGIS_feature_service_operational_layers/02qt00000017000000/
            string resourceName = @"ArteLogico.GisGlue.Tests.Contracts.SerializationTestsApprovals.SerializationTests.FeatureServiceOperationalLayersRoundtripSerialization.approved.txt";
            CheckRoundtripSerialization(resourceName);
        }

        [Test]
        public void OperationalLayerRoundTripSerialization()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/ArcGIS_map_service_operational_layer/02qt00000018000000/
            string resourceName = @"ArteLogico.GisGlue.Tests.Contracts.SerializationTestsApprovals.SerializationTests.OperationalLayerRoundTripSerialization.approved.txt";
            CheckRoundtripSerialization(resourceName);
        }

        [Test]
        public void SingleBasemapLayerRoundTripSerialization()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/Single_basemap_layer/02qt00000016000000/
            string resourceName = @"ArteLogico.GisGlue.Tests.Contracts.SerializationTestsApprovals.SerializationTests.SingleBasemapLayerRoundTripSerialization.approved.txt";
            CheckRoundtripSerialization(resourceName);
        }

        [Test]
        public void SingleOperationalLayerRoundTripSerialization()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/Single_layer_from_an_ArcGIS_map_service_as_an_operational_layer/02qt0000001m000000/
            string resourceName = @"ArteLogico.GisGlue.Tests.Contracts.SerializationTestsApprovals.SerializationTests.SingleOperationalLayerRoundTripSerialization.approved.txt";
            CheckRoundtripSerialization(resourceName);
        }

        [Test]
        public void TwoBasemapLayerRoundTripSerialization()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/Two_basemap_layers/02qt00000015000000/
            string resourceName = @"ArteLogico.GisGlue.Tests.Contracts.SerializationTestsApprovals.SerializationTests.TwoBasemapLayerRoundTripSerialization.approved.txt";
            CheckRoundtripSerialization(resourceName);
        }

        private static void CheckRoundtripSerialization(string resourceName)
        {
            string expected = Helper.ReadAllText(resourceName);
            var webMap = JsonHelper.Deserialize<WebMap>(expected);
            string result = JsonHelper.Serialize(webMap, Formatting.Indented);
            Approvals.Verify(result);
            Assert.That(result, Is.EqualTo(expected));
        }
    }
}
