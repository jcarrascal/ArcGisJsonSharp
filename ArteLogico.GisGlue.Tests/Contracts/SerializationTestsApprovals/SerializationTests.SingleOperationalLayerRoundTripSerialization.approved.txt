{
  "operationalLayers": [
    {
      "url": "http://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer/1",
      "id": "WorldTimeZones_717",
      "visibility": true,
      "opacity": 0.78,
      "mode": 1,
      "title": "World Time Zones",
      "layerDefinition": {
        "drawingInfo": {
          "renderer": {
            "type": "simple",
            "symbol": {
              "color": [ 108, 151, 203, 0 ],
              "outline": {
                "color": [ 108, 151, 203, 255 ],
                "width": 1.5,
                "type": "esriSLS",
                "style": "esriSLSSolid"
              },
              "type": "esriSFS",
              "style": "esriSFSSolid"
            }
          }
        },
        "minScale": 500000000,
        "maxScale": 0
      },
      "popupInfo": {
        "title": "Offset from UTC: {ZONE}",
        "fieldInfos": [
          {
            "fieldName": "ObjectID",
            "label": "ObjectID",
            "isEditable": false,
            "tooltip": "",
            "visible": false,
            "stringFieldOption": "textbox"
          },
          {
            "fieldName": "Shape",
            "label": "Shape",
            "isEditable": false,
            "tooltip": "",
            "visible": false,
            "stringFieldOption": "textbox"
          },
          {
            "fieldName": "ZONE",
            "label": "Time Zone",
            "isEditable": false,
            "tooltip": "",
            "visible": false,
            "format": {
              "places": 2,
              "digitSeparator": true
            },
            "stringFieldOption": "textbox"
          },
          {
            "fieldName": "SQ_MI",
            "label": "SQ_MI",
            "isEditable": false,
            "tooltip": "",
            "visible": false,
            "format": {
              "places": 2,
              "digitSeparator": true
            },
            "stringFieldOption": "textbox"
          },
          {
            "fieldName": "SQKM",
            "label": "SQKM",
            "isEditable": false,
            "tooltip": "",
            "visible": false,
            "format": {
              "places": 2,
              "digitSeparator": true
            },
            "stringFieldOption": "textbox"
          },
          {
            "fieldName": "COLORMAP",
            "label": "COLORMAP",
            "isEditable": false,
            "tooltip": "",
            "visible": false,
            "format": {
              "places": 2,
              "digitSeparator": true
            },
            "stringFieldOption": "textbox"
          }
        ],
        "description": null,
        "showAttachments": true,
        "mediaInfos": [ ]
      }
    }
  ],
  "baseMap": {
    "baseMapLayers": [
      {
        "id": "World_Light_Gray_Base_7739",
        "opacity": 1,
        "visibility": true,
        "url": "http://services.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Base/MapServer"
      },
      {
        "id": "World_Light_Gray_Reference_241",
        "isReference": true,
        "opacity": 1,
        "visibility": true,
        "url": "http://services.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Reference/MapServer"
      }
    ],
    "title": "Light Gray Canvas"
  },
  "version": "1.6"
}